esp/platform/espprotocol.cpp

```diff
@@ -831,16 +849,13 @@ class ConnectionCloseDetector : implements IConnectionCloseDetector, public CInt
     }
     virtual bool isPeerClosed()
     {
+        if (!m_isAsync && !m_isPeerClosed)
```

Comment:
Perhaps isPeerClosed() could be called both when non-async and async ?

Reply:
isPeerClosed() can be called by both sync and async. Here the check is to control whether to call checkClose(), which is not necessary for async because it's done in the async thread...

Reply:
ok, sure.